tests/testthat/test-model-tree.R

```diff
@@ -117,6 +117,43 @@ withr::with_options(list(bbr.bbi_exe_path = read_bbi_path()), {
       expect_equal(length(pl_tree$x$data$children), 2)
     })
 
+    it("Broken links - missing origin model", {
+      # This is an unlikely scenario, as the starting/origin model often remains
+      # for traceability purposes.
+      # - However, model_tree should work whenever run_log can
+
+      # New environment so origin model can be deleted
+      MODEL_DIR_BBI <- file.path(dirname(ABS_MODEL_DIR), "test-model-tree")
+      fs::dir_create(MODEL_DIR_BBI)
+      on.exit(fs::dir_delete(MODEL_DIR_BBI))
+
+      # Create origin model
+      fs::file_copy(CTL_TEST_FILE, MODEL_DIR_BBI)
+      mod1 <- new_model(file.path(MODEL_DIR_BBI, "1"))
+      copy_output_dir(MOD1, file.path(MODEL_DIR_BBI, "1"))
+
+      # Create model network with missing origin model
+      #  - broken_link is not necessary, but serves to test that a model tree
+      #    works with multiple broken links (including the origin model)
+      #  - broken_link creates another path from mod1 and deletes the first one
+      create_tree_models(mod1 = mod1, broken_link = TRUE)
+      delete_models(mod1, .tags = NULL, .force = TRUE)
+
+      run_df <- run_log(MODEL_DIR_BBI)
+
+      expect_warning(
+        tree_data <- make_tree_data(run_df, add_summary = FALSE),
+        "The following models could not be linked properly"
+      )
```

Comment:
Why do this `make_tree_data` call separately rather than use the `expect_warning` for the main `model_tree` call on line 149?  Does the cover something that doesn't?

Reply:
>Does the cover something that doesn't?

No, I guess I was just trying to "zoom in" as much as possible to show the highest level function this could be triggered on without having to manually create a custom dataset. I can make that change if you think that would be more clear.

Reply:
Up to you.  I'm good with merging this as is, but, yes, I was confused about why the inner helper was used there.  (I'm not saying it's not useful to do a lot of times, especially for exhaustively testing cases that are more  convenient to hit from the inner function.)